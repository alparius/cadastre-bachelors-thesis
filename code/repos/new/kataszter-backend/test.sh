#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
