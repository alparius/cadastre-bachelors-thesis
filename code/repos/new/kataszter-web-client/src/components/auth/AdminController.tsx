import React from "react";
import { FormattedMessage as FM } from "react-intl";
import { Header, Menu } from "semantic-ui-react";

interface IAdminControllerProps {
    activeMenu: string;
    setActiveMenu(newActiveMenu: string): void;
}

const AdminController: React.FC<IAdminControllerProps> = (props: IAdminControllerProps) => {
    const { activeMenu, setActiveMenu } = props;

    const handleMenuChange = (e: any, { name }: any) => {
        setActiveMenu(name);
    };

    return (
        <>
            <Header as="h1">
                <FM id="menu.admin" />
            </Header>
            <Menu vertical fluid secondary color="violet" size="huge">
                <Menu.Item name="all" active={activeMenu === "all"} onClick={handleMenuChange}>
                    <FM id="auth.roles.all" />
                </Menu.Item>
                <Menu.Item name="admin" active={activeMenu === "admin"} onClick={handleMenuChange}>
                    <FM id="auth.roles.admin" />
                </Menu.Item>
                <Menu.Item name="editor" active={activeMenu === "editor"} onClick={handleMenuChange}>
                    <FM id="auth.roles.editor" />
                </Menu.Item>
                <Menu.Item name="reader" active={activeMenu === "reader"} onClick={handleMenuChange}>
                    <FM id="auth.roles.reader" />
                </Menu.Item>
                <Menu.Item name="guest" active={activeMenu === "guest"} onClick={handleMenuChange}>
                    <FM id="auth.roles.guest" />
                </Menu.Item>
                <Menu.Item name="pending" active={activeMenu === "pending"} onClick={handleMenuChange}>
                    <FM id="auth.roles.pending" />
                </Menu.Item>
                <Menu.Item name="refused" active={activeMenu === "refused"} onClick={handleMenuChange}>
                    <FM id="auth.roles.refused" />
                </Menu.Item>
            </Menu>
        </>
    );
};

export default AdminController;
