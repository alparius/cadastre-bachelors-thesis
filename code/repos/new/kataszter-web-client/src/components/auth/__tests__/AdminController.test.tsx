/*===================
  AdminController.unit.test.tsx
 ===================*/
import React from "react";

import { createComponentWithIntl, mountWithIntl } from "../../../util/test/test-wrapper";
import AdminController from "../AdminController";

describe("AdminController element", () => {
    it("should render a menu", () => {
        const container = createComponentWithIntl(<AdminController activeMenu="all" setActiveMenu={jest.fn()} />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should call the setActiveMenu function when it gets clicked", () => {
        const setMenu = jest.fn();
        const container = mountWithIntl(<AdminController activeMenu="all" setActiveMenu={setMenu} />);
        const input = container.find('[name="admin"]');
        input.simulate("click");
        expect(setMenu).toHaveBeenCalledTimes(1);
    });
});
