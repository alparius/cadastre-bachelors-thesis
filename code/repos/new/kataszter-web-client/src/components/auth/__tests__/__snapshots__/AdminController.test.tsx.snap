// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminController element should render a menu 1`] = `
Array [
  <h1
    className="ui header"
  >
    Admin
  </h1>,
  <div
    className="ui violet huge fluid secondary vertical menu"
  >
    <a
      className="active item"
      onClick={[Function]}
    >
      All accounts
    </a>
    <a
      className="item"
      onClick={[Function]}
    >
      Admin
    </a>
    <a
      className="item"
      onClick={[Function]}
    >
      Editor
    </a>
    <a
      className="item"
      onClick={[Function]}
    >
      Reader
    </a>
    <a
      className="item"
      onClick={[Function]}
    >
      Guest
    </a>
    <a
      className="item"
      onClick={[Function]}
    >
      Pending
    </a>
    <a
      className="item"
      onClick={[Function]}
    >
      Refused
    </a>
  </div>,
]
`;
